components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java

```diff
@@ -2966,6 +2966,60 @@ public static NativeArray jsFunction_getSubscriptions(Context cx, Scriptable thi
         return myn;
     }
 
+    public static NativeArray jsFunction_getLightWeightAPISubscriptions(Context cx, Scriptable thisObj,
+            Object[] args, Function funObj)
+            throws ScriptException, APIManagementException {
+
+        NativeArray myn = new NativeArray(0);
+        if (args != null && 5 <= args.length) {
+            String providerName = (String) args[0];
+            String apiName = (String) args[1];
+            String version = (String) args[2];
+            String user = (String) args[3];
+            String groupingId = (String) args[4];
+
+            APIIdentifier apiIdentifier = new APIIdentifier(APIUtil.replaceEmailDomain(providerName), apiName, version);
+            Subscriber subscriber = new Subscriber(user);
+            APIConsumer apiConsumer = getAPIConsumer(thisObj);
+            Set<SubscribedAPI> apis = apiConsumer.getLightWeightSubscribedIdentifiers(subscriber, apiIdentifier,
+                                                                                      groupingId);
+            int i = 0;
+            if (apis != null) {
+                for (SubscribedAPI api : apis) {
+                    NativeObject row = new NativeObject();
+                    row.put("application", row, api.getApplication().getName());
```

Comment:
Since this is not the application object but application name how about,
```suggestion
                    row.put("applicationName", row, api.getApplication().getName());
```

Reply:
This was used to comply with the existing getSubscriptions method and thus to minimize jaggery code changes.